### PR TITLE
hugolib: Add option to disable rendering of 404 page

### LINKED ISSUE #1889: Add option to not generate a 404 page
I have no need for a 404 page.  I'd rather not add a build step to remove it if possible.

`disableRSS` and `disableSitemap` are both existing configuration variables.

https://discuss.gohugo.io/t/disable-404/2110

### PR DESCRIPTION
**Pull request by digitalcraftsman**
Fixes #1889

### CODE REVIEW

File: hugolib/site.go [hunk 1]
```diff
@@ -1800,16 +1800,18 @@ func (s *Site) RenderHomePage() error {
...
-	n.URL = helpers.URLize("404.html")
-	n.IsHome = false
-	n.Title = "404 Page not found"
-	n.Permalink = s.permalink("404.html")
-	n.scratch = newScratch()
-
-	nfLayouts := []string{"404.html"}
-	if nfErr := s.renderAndWritePage("404 page", "404.html", n, s.appendThemeTemplates(nfLayouts)...); nfErr != nil {
-		return nfErr
+	if !viper.GetBool("Disable404") {
```
Comment (bep) [inline hugolib/site.go]: It's much easier to read if you do if viper.GetBool("Disable404") return nil instead.

Comment (digitalcraftsman) [inline hugolib/site.go]: Good idea. I inverted the logic.

**Comment by bep** [comment]
Could you rebase this against master?

**Comment by digitalcraftsman** [comment]
Done

**Pull request closed without merging**